### DISCUSSION TITLE
Websocket cannot respond immediately in v0.111.0, but it can in v0.110.2.

### DISCUSSION THREAD

**Discussion by nowfox** [fastapi/fastapi#11588] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI, WebSocket, WebSocketDisconnect
from fastapi.responses import HTMLResponse
import time

app = FastAPI()

html = """
<!DOCTYPE html>
<html>
    <head>
        <title>Chat</title>
    </head>
    <body>
        <h1>WebSocket Chat</h1>
        <h2>Your ID: <span id="ws-id"></span></h2>
        <form action="" onsubmit="sendMessage(event)">
            <input type="text" id="messageText" autocomplete="off"/>
            <button>Send</button>
        </form>
        <ul id='messages'>
        </ul>
        <script>
            var client_id = Date.now()
            document.querySelector("#ws-id").textContent = client_id;
            var ws = new WebSocket(`ws://localhost:${window.location.port}/ws/${client_id}`);
            ws.onmessage = function(event) {
                var messages = document.getElementById('messages')
                var message = document.createElement('li')
                var content = document.createTextNode(event.data)
                message.appendChild(content)
                messages.appendChild(message)
            };
            function sendMessage(event) {
                var input = document.getElementById("messageText")
                ws.send(input.value)
                input.value = ''
                event.preventDefault()
            }
        </script>
    </body>
</html>
"""


class ConnectionManager:
    def __init__(self):
        self.active_connections: list[WebSocket] = []

    async def connect(self, websocket: WebSocket):
        await websocket.accept()
        self.active_connections.append(websocket)

    def disconnect(self, websocket: WebSocket):
        self.active_connections.remove(websocket)

    async def send_personal_message(self, message: str, websocket: WebSocket):
        await websocket.send_text(message)

    async def broadcast(self, message: str):
        for connection in self.active_connections:
            await connection.send_text(message)
            time.sleep(5) # new code


manager = ConnectionManager()


@app.get("/")
async def get():
    return HTMLResponse(html)


@app.websocket("/ws/{client_id}")
async def websocket_endpoint(websocket: WebSocket, client_id: int):
    await manager.connect(websocket)
    try:
        while True:
            data = await websocket.receive_text()
            await manager.send_personal_message(f"You wrote: {data}", websocket)
            await manager.broadcast(f"Client #{client_id} says: {data}")
    except WebSocketDisconnect:
        manager.disconnect(websocket)
        await manager.broadcast(f"Client #{client_id} left the chat")
```

## FastAPI 0.111.0
Open the browser and call the endpoint '/'.Enter any content in the input box, then click the **send** button.I will receive a response in 5 seconds.
## FastAPI 0.110.2
When I click the send button, I will immediately receive a response.  

I expect to receive an immediate response from websocket in the latest version.  
After I execute `pip uninstall uvloop`, it works.  
`uvloop` is installed during the installation of FastAPI.`uvloop` is a fast.But it seems to conflict with WebSocket.

### Operating System

Linux, macOS

### Operating System Details

This issue exists on both Linux and macOD.

### FastAPI Version

0.111.0

### Pydantic Version

2.7.1

### Python Version

3.9.19

**Comment by YuriiMotov** [comment]
Use `await asyncio.sleep(5)` instead of blocking `sleep(5)` inside async functions.
```
    async def broadcast(self, message: str):
        for connection in self.active_connections:
            await connection.send_text(message)
            await asyncio.sleep(5) # NOT time.sleep(5)
```

Using `time.sleep()` inside async function blocks event loop.

**Reply by nowfox** [reply]
`time.sleep(5)` is just a simulation.   In my project, it is a time-consuming operation.
When I need to send messages to multiple websockets in an async method,websocket cannot respond immediately in v0.111.0, but it can in v0.110.2.

**Reply by YuriiMotov** [reply, marked as answer]
`await asyncio.sleep(5)` also simulates time-consuming operation, but in a right way for async functions. It releases event loop (doesn't block it).

If your operation is blocking (as `time.sleep()` is) then you need either to insert some `await asyncio.sleep(0)` in this code or run this blocking code in the thread executor in order not to block event loop.

The fact that this code works on previous version and doesn't work on current version doesn't mean that current version is worse. Something was changed in the implementation and now your blocking code inside async function has more impact on the app perfomance.
But initial problem is that you use blocking code inside async function.
